### PR TITLE
Timezone per client

### PR DESCRIPTION
**Pull request by vanowm**
* Added option to change timezone per client
* Fixed teleporter incorrectly using 12 hour number in filename instead of 24 hour
* Updated Moment.js library to v2.25.3

### CODE REVIEW

**Comment by PromoFaux** [comment]
Thanks for the submission. I've asked @XhmikosR  to give it a review, because he is better with JS stuff than I...

One thing I will note though is that you are using `let` over `var`, currently we are not using es6 features, so this is why the test is failing.

**Comment by XhmikosR** [comment]
Yeah, please use plain old JS and please also drop the moment.js update; you can do that in a separate PR later. Thanks!

File: scripts/pi-hole/php/teleporter.php [hunk 1]
```diff
@@ -540,7 +540,7 @@ function process_file($contents)
 }
 else
 {
-	$tarname = "pi-hole-teleporter_".date("Y-m-d_h-i-s").".tar";
+	$tarname = "pi-hole-teleporter_".date("Y-m-d_H-i-s").".tar";
```
Comment (XhmikosR) [inline scripts/pi-hole/php/teleporter.php:543]: In fact, if you could split this patch too to a separate PR, we could get that merged sooner.

File: scripts/vendor/moment-timezone-with-data.min.js [hunk 2]
```diff
@@ -0,0 +1 @@
...Africa/Nairobi Africa/Kampala","UM|Pacific/Pago_Pago Pacific/Wake Pacific/Honolulu Pacific/Midway","US|America/New_York America/Detroit America/Kentucky/Louisville America/Kentucky/Monticello America/Indiana/Indianapolis America/Indiana/Vincennes America/Indiana/Winamac America/Indiana/Marengo America/Indiana/Petersburg America/Indiana/Vevay America/Chicago America/Indiana/Tell_City America/Indiana/Knox America/Menominee America/North_Dakota/Center America/North_Dakota/New_Salem America/North_Dakota/Beulah America/Denver America/Boise America/Phoenix America/Los_Angeles America/Anchorage America/Juneau America/Sitka America/Metlakatla America/Yakutat America/Nome America/Adak Pacific/Honolulu","UY|America/Montevideo","UZ|Asia/Samarkand Asia/Tashkent","VA|Europe/Rome Europe/Vatican","VC|America/Port_of_Spain America/St_Vincent","VE|America/Caracas","VG|America/Port_of_Spain America/Tortola","VI|America/Port_of_Spain America/St_Thomas","VN|Asia/Bangkok Asia/Ho_Chi_Minh","VU|Pacific/Efate","WF|Pacific/Wallis","WS|Pacific/Apia","YE|Asia/Riyadh Asia/Aden","YT|Africa/Nairobi Indian/Mayotte","ZA|Africa/Johannesburg","ZM|Africa/Maputo Africa/Lusaka","ZW|Africa/Maputo Africa/Harare"]}),b});
```
Comment (XhmikosR) [inline scripts/vendor/moment-timezone-with-data.min.js:1]: I'm not sure it's worth adding yet another JS file, let alone it's 191KB minified.

**Comment by yubiuser** [comment]
While your improving teleporters filename already: here is a FR to add hostname also.
https://discourse.pi-hole.net/t/hostname-in-teleporter-export/25309
